### PR TITLE
describe rationale for deprecated features

### PR DESCRIPTION
**Pull request by bashbaug**
This pull request describes the rationale for features which have been deprecated and what the replacement or equivalent feature is (when one exists).  This was a request from the OpenCL community at [IWOCL](https://www.iwocl.org/).

### CODE REVIEW

File: api/appendix_e.asciidoc [hunk 1]
```diff
@@ -156,21 +163,34 @@ The following features are added to the OpenCL C programming language
...
-  * {clUnloadCompiler} and {clGetExtensionFunctionAddress}
-  * {clCreateFromGLTexture2D} and {clCreateFromGLTexture3D}
+// Bugzilla 6597
+  * The {clEnqueueMarker}, {clEnqueueBarrier} and {clEnqueueWaitForEvents}
+    APIs are deprecated to simplify the API.
+    The {clEnqueueMarkerWithWaitList} and {clEnqueueBarrierWithWaitList}
+    APIs provide equivalent functionality and support explicit event
+    wait lists.
+// No Bugzilla
+  * The {clCreateImage2D}, {clCreateImage3D} {clCreateFromGLTexture2D} and
```
Comment (EwanC) [inline api/appendix_e.asciidoc]: comma after {clCreateImage3D}

File: api/appendix_e.asciidoc [hunk 2]
```diff
@@ -85,11 +85,19 @@ The following features are added to the OpenCL C programming language
...
+    implementations and possibly improves performance by enforcing that
+    command queue properties are invariant.
+    Applications are encouraged to create multiple command queues with
+    different properties versus modifying the properties of a single
+    command queue.
+// Bugzilla 6628
+  * The `-cl-strict-aliasing` build option has been deprecated.
+    It is no longer required after defining type-based aliasing rules.
+// Bugzilla 5593 and 6068
+  * The *cl_khr_select_fprounding_mode* extension is deprecated and its
```
Comment (AlexeySachkov) [inline api/appendix_e.asciidoc]: No much of rationale is described here :)

File: api/appendix_e.asciidoc (hunk 1 above)
Comment (bashbaug) [reply api/appendix_e.asciidoc]: fixed - thanks!

File: api/appendix_e.asciidoc [hunk 3]
```diff
@@ -219,21 +239,49 @@ The following features are added to the OpenCL C programming language
...
+    and replaced by {CL_DEVICE_QUEUE_ON_HOST_PROPERTIES}.
+// Bugzilla 8761
+  * Atomics and Fences
+  ** The Explicit Memory Fence Functions defined in section 6.12.9 of the
+     OpenCL 1.2 specification have been deprecated to simplify the
+     programming language.
+     The *atomic_work_item_fence* function provides equivalent
+     functionality.
+  ** The Atomic Functions defined in section 6.12.11 of the OpenCL 1.2
+     specification have been deprecated to simplify the programing
```
Comment (jlewis-austin) [inline api/appendix_e.asciidoc]: typo 'programing'

Comment (bashbaug) [reply api/appendix_e.asciidoc]: Fixed - thanks!

File: api/appendix_e.asciidoc (hunk 2 above)
Comment (alycm) [reply api/appendix_e.asciidoc]: I've merged this PR, this point was never resolved but I don't know what else could be said.

**Pull request merged**